### PR TITLE
Edit librustc_lexer top-level docs

### PR DESCRIPTION
**Pull request by pierwill**
Minor edit, and adds link to librustc_parse::lexer.

### CODE REVIEW

File: src/librustc_lexer/src/lib.rs [hunk 1]
```diff
@@ -6,17 +6,18 @@
...
 //! The purpose of this crate is to convert raw sources into a labeled sequence
 //! of well-known token types, so building an actual Rust token stream will
 //! be easier.
 //!
-//! Main entity of this crate is [`TokenKind`] enum which represents common
+//! The main entity of this crate is the [`TokenKind`] enum which represents common
 //! lexeme types.
-
+//!
+//! [`librustc_parse::lexer`]: https://doc.rust-lang.org/nightly/nightly-rustc/rustc_parse/lexer/index.html
```
Comment (pierwill) [inline src/librustc_lexer/src/lib.rs:20]: Is this the best way to link to docs across crates?

Comment (jonas-schievink) [reply src/librustc_lexer/src/lib.rs:20]: Hmm, does a relative link not work? Or rustdoc's intra-doc-links?

```suggestion
//! [`librustc_parse::lexer`]: ../rustc_parse/lexer/index.html
```

Comment (pierwill) [reply src/librustc_lexer/src/lib.rs:20]: Ah! I figured out how to test this locally with two `cargo doc --open` commands, one in each crate.

This works 👍

Comment (pierwill) [reply src/librustc_lexer/src/lib.rs]: Fixed.

**Pull request merged**